pennylane/devices/experimental/device_api.py

```diff
@@ -0,0 +1,555 @@
+
+        :meth:`~.Device.batch_execute` and :meth:`~.Device.execute` are now a single method, :meth:`~.Device.execute`
+
+        :meth:`~.Device.batch_transform` and :meth:`~.Device.expand_fn` are now a single method, :meth:`~.Device.preprocess`
+
+        Shot information is no longer stored on the device, but instead specified on individual input :class:`~.QuantumTape`.
+
+        The old devices defined a :meth:`~.Device.capabilities` dictionary that defined characteristics of the devices and controlled various
+        preprocessing and validation steps, such as ``"supports_broadcasting"``.  These capabilites should now be handled by the
+        :meth:`~.Device.preprocess` method. For example, if a device does not support broadcasting, ``preprocess`` should
+        split a quantum script with broadcasted parameters into a batch of quantum scripts. If the device does not support mid circuit
+        measurements, then ``preprocess`` should apply :func:`~.defer_measurements`.  A set of default preprocessing steps will be available
+        to make a seemless transition to the new interface.
+
+        A class will be provided to easily construct default preprocessing steps from supported operations, supported observables,
+        supported measurement processes, and various capabilities.
+
+        Utility functions will be added to the ``devices`` module to query whether or not the device driver can do certain things, such
+        as ``devices.supports_operator(op, dev, native=True)``. These functions will work by checking the behaviour of :meth:`~.Device.preprocess`
+        to certain inputs.
+
+        Versioning should be specified by the package containing the device. If an external package includes a PennyLane device,
+        then the package requirements should specify the minimium PennyLane version required to work with the device.
+
+
+    .. details::
+        :title: Execution Configuration
+
+        Execution config properties most relevant for devices are:
+
+        * ``device_options``: A dictionary of device specific options. For example, the python device may have ``multiprocessing_mode``
+          as a key. These should be documented in the class docstring.
+
+        * ``gradient_method``: A device can choose to have native support for any type of gradient method. If the method
+          :meth:`~.supports_derivatives` returns ``True`` for a particular gradient method, it will be treated as a device
+          derivative and not handled by pennylane core code.
+
+        * ``gradient_keyword_arguments``: Options for the gradient method.
+
+        * ``derivative_order``: Relevant for requested device derivatives.
+
+    """
+
+    @property
+    def name(self) -> str:
+        """The name of the device or set of devices.
+
+        This property can either be the name of the class, or an alias to be used in the :func:`~.device` constructor,
+        such as ``"default.qubit"`` or ``"lightning.qubit"``.
```

Comment:
we'll have to remember to change this example (assuming those names do not carry over)

Reply:
Another detail to clarify when we have an actual implementation to demonstrate with.